tests/common/PlatformInfo.cs

```diff
@@ -91,92 +119,75 @@ static PlatformInfo GetHostPlatformInfo ()
 
 	public static class AvailabilityExtensions
 	{
-#if NET
-		public static AvailabilityBaseAttribute Convert (this OSPlatformAttribute a)
-		{
-			if (a == null)
-				return null;
-
-			PlatformName p;
-			int n = 0;
-			switch (a.PlatformName) {
-			case string dummy when a.PlatformName.StartsWith ("ios"):
-				p = PlatformName.iOS;
-				n = "ios".Length;
-				break;
-			case string dummy when a.PlatformName.StartsWith ("tvos"):
-				p = PlatformName.TvOS;
-				n = "tvos".Length;
-				break;
-			case string dummy when a.PlatformName.StartsWith ("watchos"):
-				p = PlatformName.WatchOS;
-				n = "watchos".Length;
-				break;
-			case string dummy when a.PlatformName.StartsWith ("macos"):
-				p = PlatformName.MacOSX;
-				n = "macos".Length;
-				break;
-			case string dummy when a.PlatformName.StartsWith ("maccatalyst"):
-				p = PlatformName.MacCatalyst;
-				n = "maccatalyst".Length;
-				break;
-			default:
-				return null;
-			}
-			bool versioned = Version.TryParse (a.PlatformName [n..], out var v);
-
-			if (a is SupportedOSPlatformAttribute) {
-				if (!versioned)
-					throw new FormatException (a.PlatformName);
-				return new IntroducedAttribute (p, v.Major, v.Minor);
-			}
-			if (a is UnsupportedOSPlatformAttribute) {
-				// if a version is provided then it means it was deprecated
-				// if no version is provided then it means it's unavailable
-				if (versioned)
-					return new DeprecatedAttribute (p, v.Major, v.Minor);
-				else
-					return new UnavailableAttribute (p);
-			}
-			return null;
-		}
-#endif
-
 		public static bool IsAvailableOnHostPlatform (this ICustomAttributeProvider attributeProvider)
 		{
 			return attributeProvider.IsAvailable (PlatformInfo.Host);
 		}
 
 		public static bool IsAvailable (this ICustomAttributeProvider attributeProvider, PlatformInfo targetPlatform)
 		{
+			var customAttributes = attributeProvider.GetCustomAttributes (true);
+
 #if NET
-			var list = new List<AvailabilityBaseAttribute> ();
-			foreach (var ca in attributeProvider.GetCustomAttributes (true)) {
-				if (ca is OSPlatformAttribute aa) {
-					var converted = aa.Convert ();
-					if (converted is not null)
-						list.Add (converted);
-				}
-				// FIXME - temporary, while older attributes co-exists (in manual bindings)
-				if (ca is AvailabilityBaseAttribute old)
-					list.Add (old);
-				if (ca is ObsoleteAttribute)
-					return false;
-			}
-			return list.IsAvailable (targetPlatform);
+			customAttributes = customAttributes.ToArray (); // don't iterate twice
+			if (customAttributes.Any (v => v is ObsoleteAttribute))
```

Comment:
LINQ here is making the implementation more complex than a normal foreach loop :D